src/main/groovy/edu/oregonstate/mist/personsapi/mapper/PersonMapper.groovy

```
@@ -14,6 +14,17 @@ import java.sql.SQLException
 public class PersonMapper implements ResultSetMapper<PersonObject> {
     PhoneFormatter phoneFormatter = new PhoneFormatter()
     public PersonObject map(int i, ResultSet rs, StatementContext sc) throws SQLException {
+        String rawSSN = rs.getString('SSN')
+        String ssnStatus = 'invalid'
+
+        if (rawSSN && rawSSN.startsWith('VAULT')) {
+            ssnStatus = 'valut'
+        } else if (rawSSN == 'REDACTED' || rawSSN == 'ARCHIVED') {
+            ssnStatus = rawSSN.toLowerCase()
+        } else if (rawSSN && rawSSN.matches(/^(?!0{9})\d{9}$/)) {
+            ssnStatus = 'valid'
+        }
```

What should happen if `rawSSN` doesn't match any of the cases here? Should we throw an exception or something?

---

No, then the `ssnStatus` should be `invalid`.